### PR TITLE
[ingress-nginx] Add the loadBalancerClass setting to IngressNginxController resource

### PR DESCRIPTION
**Pull request by tarabrind**
## Description
- The `loadBalancerClass` parameter has been added to the `IngressNginxController` resource.
- The field is fully passed to the `<name>-load-balancer` Service as `.spec.loadBalancerClass`.

## Why do we need it, and what problem does it solve?
There could be several load-balancer controllers in cluster (i.e. when using several MetalLoadBalancerClass in metallb #9658).

## What is the expected result?
When setting the `IngressNginxController` parameter `.spec.<inlet>.loadBalancerClass` it passes to the `.spec.loadBalancerClass` of `<name>-load-balancer` Service.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: ingress-nginx
type: chore
summary: Added an optional loadBalancerClass setting to IngressNginxController resource.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @tarabrind. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 1]
```diff
@@ -168,6 +168,10 @@ spec:
                         Note that you will need to recreate `IngressNginxController` (or create a new controller and then delete the old one) if annotations to provision a load balancer are only used when creating the service.
                       additionalProperties:
                         type: string
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        Allows you to modify the `.spec.loadBalancerClass` setting of a standard Service object.
```
Comment (Beastlex) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:174]:
```suggestion
                        Allows you to modify the `.spec.loadBalancerClass` setting of a ingress-managed load balancer.
```

**Review by Beastlex — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Review by AndreyPavlovFlant — APPROVED** [review]

File: modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml [hunk 2]
```diff
@@ -94,6 +94,9 @@ spec:
 
                         **Внимание!** Модуль не учитывает особенности указания аннотаций в различных облаках.
                         Если аннотации для заказа LoadBalancer'а применяются только при создании сервиса, для обновления подобных параметров необходимо будет пересоздать `IngressNginxController` (или создать новый, затем удалив старый).
+                    loadBalancerClass:
+                      description: |
+                        Позволяет изменять настройку `.spec.loadBalancerClass` балансировщика нагрузки, управляемого с помощью Ingress.
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml:99]:
```suggestion
                        Класс балансировщика входящих сетевых запросов (пробрасывается в параметр `spec.loadBalancerClass` заказанного сервиса с типом LoadBalancer).
```

File: modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml [hunk 3]
```diff
@@ -122,6 +125,9 @@ spec:
                         Аннотации, которые будут проставлены сервису для гибкой настройки балансировщика.
 
                         **Внимание!** Модуль не учитывает особенности указания аннотаций в различных облаках. Если аннотации для заказа LoadBalancer'а применяются только при создании сервиса, для обновления подобных параметров необходимо будет пересоздать `IngressNginxController` (или создать новый, затем удалив старый).
+                    loadBalancerClass:
+                      description: |
+                        Позволяет изменять настройку `.spec.loadBalancerClass` балансировщика нагрузки, управляемого с помощью Ingress.
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml:130]:
```suggestion
                        Класс балансировщика входящих сетевых запросов (пробрасывается в параметр `spec.loadBalancerClass` заказанного сервиса с типом LoadBalancer).
```

File: modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml [hunk 4]
```diff
@@ -451,6 +457,9 @@ spec:
 
                         > **Внимание!** Модуль не учитывает особенности указания аннотаций в различных облаках.
                         > Если аннотации для заказа LoadBalancer'а применяются только при создании сервиса, для обновления подобных параметров необходимо будет пересоздать `IngressNginxController` (или создать новый, затем удалив старый).
+                    loadBalancerClass:
+                      description: |
+                        Позволяет изменять настройку `.spec.loadBalancerClass` балансировщика нагрузки, управляемого с помощью Ingress.
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml:462]:
```suggestion
                        Класс балансировщика входящих сетевых запросов (пробрасывается в параметр `spec.loadBalancerClass` заказанного сервиса с типом LoadBalancer).
```

File: modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml [hunk 5]
```diff
@@ -478,6 +487,9 @@ spec:
                         Аннотации, которые будут проставлены сервису для гибкой настройки балансировщика.
 
                         > **Внимание!** Модуль не учитывает особенности указания аннотаций в различных облаках. Если аннотации для заказа LoadBalancer'а применяются только при создании сервиса, для обновления подобных параметров необходимо будет пересоздать `IngressNginxController` (или создать новый, затем удалив старый).
+                    loadBalancerClass:
+                      description: |
+                        Позволяет изменять настройку `.spec.loadBalancerClass` балансировщика нагрузки, управляемого с помощью Ingress.
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml:492]:
```suggestion
                        Класс балансировщика входящих сетевых запросов (пробрасывается в параметр `spec.loadBalancerClass` заказанного сервиса с типом LoadBalancer).
```

File: modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml [hunk 6]
```diff
@@ -495,6 +507,9 @@ spec:
...
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml:512]:
```suggestion
                        Класс балансировщика входящих сетевых запросов (пробрасывается в параметр `spec.loadBalancerClass` заказанного сервиса с типом LoadBalancer).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 7]
```diff
@@ -168,6 +168,10 @@ spec:
                         Note that you will need to recreate `IngressNginxController` (or create a new controller and then delete the old one) if annotations to provision a load balancer are only used when creating the service.
                       additionalProperties:
                         type: string
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        Allows you to modify the `.spec.loadBalancerClass` setting of a ingress-managed load balancer.
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:174]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 8]
```diff
@@ -210,6 +214,10 @@ spec:
                       additionalProperties:
                         type: string
                       x-kubernetes-preserve-unknown-fields: true
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        Allows you to modify the `.spec.loadBalancerClass` setting of a ingress-managed load balancer.
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:220]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 9]
```diff
@@ -752,6 +760,10 @@ spec:
                         Note that you will need to recreate `IngressNginxController` (or create a new controller and then delete the old one) if annotations to provision a load balancer are only used when creating the service.
                       additionalProperties:
                         type: string
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        Allows you to modify the `.spec.loadBalancerClass` setting of a ingress-managed load balancer.
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:766]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 10]
```diff
@@ -794,6 +806,10 @@ spec:
                       additionalProperties:
                         type: string
                       x-kubernetes-preserve-unknown-fields: true
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        Allows you to modify the `.spec.loadBalancerClass` setting of a ingress-managed load balancer.
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:812]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 11]
```diff
@@ -822,6 +838,10 @@ spec:
...
```
Comment (apolovov) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:844]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```

**Review by apolovov — CHANGES_REQUESTED** [review]

**Review by apolovov — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 12]
```diff
@@ -168,6 +168,10 @@ spec:
                         Note that you will need to recreate `IngressNginxController` (or create a new controller and then delete the old one) if annotations to provision a load balancer are only used when creating the service.
                       additionalProperties:
                         type: string
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```
Comment (kirkonru) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:174]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service of the LoadBalancer type).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 13]
```diff
@@ -210,6 +214,10 @@ spec:
                       additionalProperties:
                         type: string
                       x-kubernetes-preserve-unknown-fields: true
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```
Comment (kirkonru) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:220]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service of the LoadBalancer type).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 14]
```diff
@@ -752,6 +760,10 @@ spec:
                         Note that you will need to recreate `IngressNginxController` (or create a new controller and then delete the old one) if annotations to provision a load balancer are only used when creating the service.
                       additionalProperties:
                         type: string
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```
Comment (kirkonru) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:766]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service of the LoadBalancer type).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 15]
```diff
@@ -794,6 +806,10 @@ spec:
                       additionalProperties:
                         type: string
                       x-kubernetes-preserve-unknown-fields: true
+                    loadBalancerClass:
+                      type: string
+                      description: |
+                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service with the LoadBalancer type).
```
Comment (kirkonru) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:812]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service of the LoadBalancer type).
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 16]
```diff
@@ -822,6 +838,10 @@ spec:
...
```
Comment (kirkonru) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:844]:
```suggestion
                        The LoadBalancer class (it is passed to the `spec.loadBalancerClass` parameter of the provisioned service of the LoadBalancer type).
```

**Review by kirkonru — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [9665](https://github.com/deckhouse/deckhouse/pull/9665) to the branch [release-1.64](https://github.com/deckhouse/deckhouse/tree/release-1.64) successful!

**Pull request merged**